File: Java_Design-Patterns/src/composite/Contenedor.java
package composite;

import java.util.ArrayList;
import java.util.List;

public class Contenedor implements Componente {

    private String nombre;
    private List<Componente> contenedor;

    public Contenedor() {
        this.nombre = "";
        this.contenedor = new ArrayList<>();
    }

    public Contenedor(String nombre) {
        this.nombre = nombre;
        this.contenedor = new ArrayList<>();
    }

    @Override
    public void pintar(String color) {
        for (Componente componente : this.contenedor) {
            componente.pintar(color);
        }
        System.out.println("Se pinto a " + nombre + " de color " + color);
    }

    public List<Componente> getContenedor() {
        return contenedor;
    }

    public void setContenedor(List<Componente> contenedor) {
        this.contenedor = contenedor;
    }

    public void add(Componente componente) {
        this.contenedor.add(componente);
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

}

File: Python_Design-Patterns/composite/contenedor.py
from composite.componente import Componente


class Contenedor(Componente):

    def __init__(self, nombre):
        self._nombre = nombre
        self._contenedor = []

    def pintar(self, color):
        for Componente in self._contenedor:
            Componente.pintar(color)
        print(f'Se pinto a {self.nombre} de color {color}')

    @property
    def contenedor(self):
        return self._contenedor

    @contenedor.setter
    def contenedor(self, contenedor):
        self.contenedor = contenedor

    def add(self, Componente):
        self._contenedor.append(Componente)

    @property
    def nombre(self):
        return self._nombre

    @nombre.setter
    def nombre(self, nombre):
        self._nombre = nombre
